### PR TITLE
Improve plugin performance

### LINKED ISSUE #2837: [Performance Improvement] Improve SolrJ Agent Plugin's performance
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [ ] Bug
- [ ] Requirement
- [x] Feature or performance improvement

___
### Requirement or improvement

- Please describe about your requirements or improvement suggestions.

https://github.com/apache/skywalking/blob/7f2277091dcab2ee7c2cabb414622d4fd99064e8/apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/solrj/SolrClientInterceptor.java#L195-L201

the two methods use `String.format` to concat strings, and in my memory, `String.format` is  more expensive than `+` operator. the two methods would be called very frequently in real scenario, so I think improving this is not trivial. What do you think @wu-sheng ?

> in the code I posted above, replacing the `String.format` with `+` operator won't effect the readability too much, that's why I think it's worth improving.

Below is my benchmark and result:

```java
@BenchmarkMode(Mode.Throughput)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
@State(Scope.Thread)
@Fork(2)
@Warmup(iterations = 4)
@Measurement(iterations = 5)
public class StringFormatBenchmarkTest {
    @Benchmark
    public void testStringFormat() {
        for (int i = 0; i < 100; i++) {
            String.format("solrJ/%s%s/%s", i, i, i);
        }
    }

    @Benchmark
    public void testStringConcat() {
        for (int i = 0; i < 100; i++) {
            String a = "solrJ/" + i + i + "/" + i;
        }
    }

    @BenchmarkMode(Mode.AverageTime)
    @OutputTimeUnit(TimeUnit.MICROSECONDS)
    public static void main(String[] args) throws RunnerException {
        Options options = new OptionsBuilder().include(StringFormatBenchmarkTest.class.getSimpleName()).build();
        new Runner(options).run();
    }
}
```

```text
# JMH version: 1.21
# VM version: JDK 1.8.0_181, Java HotSpot(TM) 64-Bit Server VM, 25.181-b13
# VM invoker: /Library/Java/JavaVirtualMachines/jdk1.8.0_181.jdk/Contents/Home/jre/bin/java
# VM options: -javaagent:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=51939:/Applications/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8
# Warmup: 4 iterations, 10 s each
# Measurement: 5 iterations, 10 s each
# Timeout: 10 min per iteration
# Threads: 1 thread, will synchronize iterations
# Benchmark mode: Throughput, ops/time
# Benchmark: org.apache.skywalking.apm.plugin.solrj.StringFormatBenchmark.testStringConcat
```

## Result

```
Benchmark                                Mode  Cnt    Score    Error   Units
StringFormatBenchmark.testStringConcat  thrpt   10  326.444 ± 46.432  ops/ms
StringFormatBenchmark.testStringFormat  thrpt   10    6.094 ±  1.065  ops/ms
```

### PR DESCRIPTION
**Pull request by kezhenxu94**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [x] Improve performance

- Related issues

fixes #2837 

___
### New feature or improvement

- Describe the details and related test reports.

replace `String.format` with `+` operator

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/test/java/org/apache/skywalking/apm/plugin/solrj/StringFormatBenchmarkTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author kezhenxu94
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/test/java/org/apache/skywalking/apm/plugin/solrj/StringFormatBenchmarkTest.java]: Please post the benchmark result here.

Comment (kezhenxu94) [reply apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/test/java/org/apache/skywalking/apm/plugin/solrj/StringFormatBenchmarkTest.java]: OK

**Comment by wu-sheng** [comment]
Also, you CI is failing, please fix it.

**Comment by kezhenxu94** [comment]
> Also, you CI is failing, please fix it.

It's weird that it failed on OracleJDK but passed on OpenJDK, now the master branch failed too. I need to do some research on this

**Comment by wu-sheng** [comment]
Because of `configuration-nacos`?

**Comment by wu-sheng** [comment]
@kezhenxu94 See this, https://travis-ci.org/apache/skywalking/builds/543132461 . 
Look like nacos test is not stable,  read log https://travis-ci.org/apache/skywalking/jobs/543132463

We need to figure some method or remove this test case somehow.

**Comment by wu-sheng** [comment]
> [ERROR] DOCKER> [nacos/nacos-server:1.0.0] "nacos-dynamic-configuration-integration-test-nacos": Timeout after 60293 ms while waiting on log out 'Nacos started successfully'
[ERROR] DOCKER> Error occurred during container startup, shutting down...
[ERROR] DOCKER> I/O Error [[nacos/nacos-server:1.0.0] "nacos-dynamic-configuration-integration-test-nacos": Timeout after 60293 ms while waiting on log out 'Nacos started successfully']
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.28.0:start (start) on project configuration-nacos: I/O Error: [nacos/nacos-server:1.0.0] "nacos-dynamic-configuration-integration-test-nacos": Timeout after 60293 ms while waiting on log out 'Nacos started successfully' -> [Help 

Nacos test starts up slowly.

**Comment by kezhenxu94** [comment]
> > [ERROR] DOCKER> [nacos/nacos-server:1.0.0] "nacos-dynamic-configuration-integration-test-nacos": Timeout after 60293 ms while waiting on log out 'Nacos started successfully'
> > [ERROR] DOCKER> Error occurred during container startup, shutting down...
> > [ERROR] DOCKER> I/O Error [[nacos/nacos-server:1.0.0] "nacos-dynamic-configuration-integration-test-nacos": Timeout after 60293 ms while waiting on log out 'Nacos started successfully']
> > [ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.28.0:start (start) on project configuration-nacos: I/O Error: [nacos/nacos-server:1.0.0] "nacos-dynamic-configuration-integration-test-nacos": Timeout after 60293 ms while waiting on log out 'Nacos started successfully' -> [Help
> 
> Nacos test starts up slowly.

It passed again, I'll try to increase the timeout

File: apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/test/java/org/apache/skywalking/apm/plugin/solrj/StringFormatBenchmarkTest.java (hunk 1 above)
Comment (kezhenxu94) [reply apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/test/java/org/apache/skywalking/apm/plugin/solrj/StringFormatBenchmarkTest.java]: Added

**Comment by kezhenxu94** [comment]
> @kezhenxu94 See this, https://travis-ci.org/apache/skywalking/builds/543132461 .
> Look like nacos test is not stable, read log https://travis-ci.org/apache/skywalking/jobs/543132463
> 
> We need to figure some method or remove this test case somehow.

The problem that Nacos starts timeout has been resolved because the database is not initialized properly, now the reason why it fails became stable:

```text
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:2.22.0:verify (default) on project configuration-nacos: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/travis/build/apache/skywalking/oap-server/server-configuration/configuration-nacos/target/failsafe-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/travis/build/apache/skywalking/oap-server/server-configuration/configuration-nacos && /usr/lib/jvm/java-8-oracle/jre/bin/java -javaagent:/home/travis/.m2/repository/org/jacoco/org.jacoco.agent/0.8.3/org.jacoco.agent-0.8.3-runtime.jar=destfile=/home/travis/build/apache/skywalking/oap-server/server-configuration/configuration-nacos/target/jacoco.exec -jar /home/travis/build/apache/skywalking/oap-server/server-configuration/configuration-nacos/target/surefire/surefirebooter4815176584272407938.jar /home/travis/build/apache/skywalking/oap-server/server-configuration/configuration-nacos/target/surefire 2019-06-08T16-52-43_685-jvmRun1 surefire9151435006599987077tmp surefire_13688043696340956275tmp
[ERROR] Process Exit Code: 0

// more stack trace ...
```

After Googling for a while, I cannot figure out why it happens yet.

**Comment by kezhenxu94** [comment]
It sill failed occasionally, and I did not figure out why after struggling half a day, I'm afraid that I have to remove the test case temporarily

**Comment by wu-sheng** [comment]
Then, I think we should remove it. Do you know jenkins in better way? I want to move more ci to ASF jenkins, are you interesting?

**Comment by kezhenxu94** [comment]
> Then, I think we should remove it.

I'll send another PR to remove the test temporarily.

> Do you know jenkins in better way? I want to move more ci to ASF jenkins, are you interesting?

I'm interested in this though I know little about Jenkins, but as far as I know in the Apache Dubbo Mailing List, configuring Jenkins job in ASF's infra needs more privileges, will do some research about Jenkins CI first and ping you

**Comment by wu-sheng** [comment]
I have the privilege, as I am project vp and apache officer. Send me your Apache id private, I will grant that to you.

**Pull request merged**